ec/ecresource/deploymentresource/read.go

```diff
@@ -180,6 +181,33 @@ func (r *Resource) read(ctx context.Context, id string, state *deploymentv2.Depl
 		deployment.Elasticsearch.Strategy = baseElasticsearch.Strategy.ValueStringPointer()
 	}
 
+	if baseElasticsearch != nil && deployment.Elasticsearch != nil && !baseElasticsearch.KeystoreContents.IsNull() {
```

Comment:
Can we add a comment on this block so it's easier to understand the intent here. IIUC:

* Read TF state into `deployment.Elasticsearch.KeystoreContents`
* Get the current keystore from the API
* Any keystore items in TF state but not in the API response are removed

The above ensures that:
* New keystore items in TF config are added
* Keystore items not managed by TF are left alone
* Keystore items previously managed, but removed from config are removed

Is the above correct?

Reply:
Thank you for the question - I'll add a comment.